File: backend-node/src/services/WeatherService.js
class WeatherService {
    static async getCurrentTemperature(latitude = 35.6895, longitude = 139.6917) {
        try {
            // Tokyo Coordinates by default
            const url = `https://api.open-meteo.com/v1/forecast?latitude=${latitude}&longitude=${longitude}&current_weather=true`;
            const response = await fetch(url);

            if (!response.ok) {
                console.warn("[Weather] API Error:", response.statusText);
                return null;
            }

            const data = await response.json();
            return data.current_weather ? data.current_weather.temperature : null;
        } catch (e) {
            console.warn("[Weather] Fetch failed:", e.message);
            return null;
        }
    }
}

module.exports = WeatherService;

File: backend-rails/app/services/weather_service.rb
require 'net/http'
require 'json'

class WeatherService
  # Kanazawa Coordinates or approximate default
  LAT = 36.56
  LON = 136.66

  def self.get_current_temperature
    url = URI("https://api.open-meteo.com/v1/forecast?latitude=#{LAT}&longitude=#{LON}&current_weather=true")
    
    begin
      response = Net::HTTP.get(url)
      data = JSON.parse(response)
      
      if data && data['current_weather']
        return data['current_weather']['temperature']
      end
    rescue => e
      puts "[Weather] Error: #{e.message}"
    end
    
    nil
  end
end
